### DISCUSSION TITLE
Class "ZipArchive" not found

### DISCUSSION THREAD

**Discussion by holyboly** [spatie/laravel-backup#1761] [Q&A]
Version: spatie/laravel-backup 8.6.0

Happened after composer update to the newest version.

/vendor/spatie/laravel-backup/config/backup.php

    'compression_method' => ZipArchive::CM_DEFAULT,

Exception:

    [2024-02-16 12:41:50] local.ERROR: Class "ZipArchive" not found {"exception":"[object] (Error(code: 0): Class \"ZipArchive\" not found at /project_dir/vendor/spatie/laravel-backup/config/backup.php:128)
    [stacktrace]
    #0 /project_dir/vendor/laravel/framework/src/Illuminate/Support/ServiceProvider.php(138): require()
    #1 /project_dir/vendor/spatie/laravel-package-tools/src/PackageServiceProvider.php(33): Illuminate\\Support\\ServiceProvider->mergeConfigFrom()
    #2 /project_dir/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(788): Spatie\\LaravelPackageTools\\PackageServiceProvider->register()
    #3 /project_dir/vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(75): Illuminate\\Foundation\\Application->register()
    #4 /project_dir/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(765): Illuminate\\Foundation\\ProviderRepository->load()
    #5 /project_dir/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/RegisterProviders.php(17): Illuminate\\Foundation\\Application->registerConfiguredProviders()
    #6 /project_dir/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(263): Illuminate\\Foundation\\Bootstrap\\RegisterProviders->bootstrap()
    #7 /project_dir/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(447): Illuminate\\Foundation\\Application->bootstrapWith()
    #8 /project_dir/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(199): Illuminate\\Foundation\\Console\\Kernel->bootstrap()
    #9 Command line code(1): Illuminate\\Foundation\\Console\\Kernel->handle()
    #10 {main}
    "}

**Comment by vdbelt** [comment]
Do you have the zip extension available?

**Comment by holyboly** [comment]
Yes, this code runs without problems:

    Artisan::command('dev:test', function () {
        $zip = new \ZipArchive();
        $this->info('no error');
    });

**Comment by albertovalerio** [comment]
Same problem here on Ubuntu 22.04, in my case I solved by downgrading php from 8.3 to 8.2 and reinstalling the zip extension.

**Comment by websitevirtuoso** [comment]
Does it mean that it can't work with php8.3?
I am getting the same problem with installed zip extension

**Comment by websitevirtuoso** [comment]
![image](https://github.com/spatie/laravel-backup/assets/5112918/d8cab6a1-f88d-48b5-a0c1-ba1766b021f0)

**Comment by holyboly** [comment]
I am using the php docker image `FROM php:8.1.16-apache`, so it does not seem to be an PHP 8.3 issue?
Not sure why, but the class `ZipArchive` cannot be found, even if I add a namespace to the config file `/vendor/spatie/laravel-backup/config/backup.php`, like `'compression_method' => \ZipArchive::CM_DEFAULT,`.

**Comment by ayuniptrii2** [comment]
i have same problem here. but in my proble, i try upgrade my laravel from 8 to 10 instead 8 to 9 to 10

**Comment by prostoleo** [comment]
Got same problem on Windows. Fixed by enabling zip extension in `php.ini` file.

```
extension=./ext/php_zip.dll
```

**Comment by TheArKaID** [comment]
Just happened to me. In Laravel 10 (upgraded from 8, 9), the extension is already enabled, using the Laravel Homestead environment, so I'm not sure how it goes. But it happened after installing PHP 8.3, just a guess because this installation was the last modification I did.

**Comment by headyj** [comment]
Had the same issue while migrating from Laravel 9 to 10 and using PHP 8.3. Just fixed it by adding libzip-dev and installing the extension:
```
apt update && apt install -y libzip-dev
docker-php-ext-install zip
```
